### ISSUE TITLE
Internal Postal Error v2.1.4

### ISSUE THREAD

**Issue by erichk4** [postalserver/postal#2451]
## Describe the bug

Incoming/Outgoing Messages marked as spam will raise an Internal Postal Error

## Screenshots

![2023-05-08_15h11_54](https://user-images.githubusercontent.com/4427908/236833206-87507f04-3664-430c-ac9a-4b4ac26ccb23.png)

## Environment details 

 - Version 2.1.4

## Additional information/context

```ruby
[2023-05-08T13:30:03.060] INFO -- : [MHLK431S2K] Started processing UnqueueMessageJob job
[2023-05-08T13:30:03.070] INFO -- : [MHLK431S2K] Lock acquired for queued message 49488
[2023-05-08T13:30:03.084] INFO -- : [MHLK431S2K] Found 0 associated messages to process at the same time (batch key: outgoing-xxx.at)
[2023-05-08T13:30:03.089] INFO -- : [MHLK431S2K] [2::49443 49488] Got queued message with exclusive lock
[2023-05-08T13:30:03.101] INFO -- : [MHLK431S2K] [2::49443 49488] Recipient is on the suppression list. Holding.
[2023-05-08T13:30:03.112] INFO -- : [MHLK431S2K] [2::49443 49488] Internal error: FrozenError: can't modify frozen String: ""
[2023-05-08T13:30:03.112] INFO -- : [MHLK431S2K] [2::49443 49488] /opt/postal/app/lib/postal/message_db/delivery.rb:51:in "force_encoding"
[2023-05-08T13:30:03.112] INFO -- : [MHLK431S2K] [2::49443 49488] /opt/postal/app/lib/postal/message_db/delivery.rb:51:in "webhook_hash"
[2023-05-08T13:30:03.112] INFO -- : [MHLK431S2K] [2::49443 49488] /opt/postal/app/lib/postal/message_db/delivery.rb:43:in "send_webhooks"
[2023-05-08T13:30:03.113] INFO -- : [MHLK431S2K] [2::49443 49488] /opt/postal/app/lib/postal/message_db/delivery.rb:11:in "create"
[2023-05-08T13:30:03.113] INFO -- : [MHLK431S2K] [2::49443 49488] /opt/postal/app/lib/postal/message_db/message.rb:124:in "create_delivery"
[2023-05-08T13:30:03.113] INFO -- : [MHLK431S2K] [2::49443 49488] /opt/postal/app/app/jobs/unqueue_message_job.rb:310:in "block in perform"
[2023-05-08T13:30:03.113] INFO -- : [MHLK431S2K] [2::49443 49488] /opt/postal/app/app/jobs/unqueue_message_job.rb:31:in "each"
[2023-05-08T13:30:03.113] INFO -- : [MHLK431S2K] [2::49443 49488] /opt/postal/app/app/jobs/unqueue_message_job.rb:31:in "perform"
[2023-05-08T13:30:03.113] INFO -- : [MHLK431S2K] [2::49443 49488] /opt/postal/app/lib/postal/worker.rb:64:in "receive_job"
[2023-05-08T13:30:03.113] INFO -- : [MHLK431S2K] [2::49443 49488] /opt/postal/app/lib/postal/worker.rb:96:in "block in join_queue"
[2023-05-08T13:30:03.113] INFO -- : [MHLK431S2K] [2::49443 49488] /usr/local/bundle/gems/bunny-2.20.3/lib/bunny/consumer.rb:56:in "call"
[2023-05-08T13:30:03.113] INFO -- : [MHLK431S2K] [2::49443 49488] /usr/local/bundle/gems/bunny-2.20.3/lib/bunny/channel.rb:1842:in "block in handle_frameset"
[2023-05-08T13:30:03.113] INFO -- : [MHLK431S2K] [2::49443 49488] /usr/local/bundle/gems/bunny-2.20.3/lib/bunny/consumer_work_pool.rb:108:in "block (2 levels) in run_loop"
[2023-05-08T13:30:03.113] INFO -- : [MHLK431S2K] [2::49443 49488] /usr/local/bundle/gems/bunny-2.20.3/lib/bunny/consumer_work_pool.rb:103:in "loop"
[2023-05-08T13:30:03.114] INFO -- : [MHLK431S2K] [2::49443 49488] /usr/local/bundle/gems/bunny-2.20.3/lib/bunny/consumer_work_pool.rb:103:in "block in run_loop"
[2023-05-08T13:30:03.114] INFO -- : [MHLK431S2K] [2::49443 49488] /usr/local/bundle/gems/bunny-2.20.3/lib/bunny/consumer_work_pool.rb:102:in "catch"
[2023-05-08T13:30:03.114] INFO -- : [MHLK431S2K] [2::49443 49488] /usr/local/bundle/gems/bunny-2.20.3/lib/bunny/consumer_work_pool.rb:102:in "run_loop"
[2023-05-08T13:30:03.118] INFO -- : [MHLK431S2K] [2::49443 49488] Queued message was unlocked
``` 

Never seen this error on v2.1.2

Maybe a double of https://github.com/postalserver/postal/issues/2153

**Comment by erichk4** [comment]
@odoo-sh  Hi,
as a "workaround" i  did the following in:

postal/lib/postal/message_db/delivery.rb:

`def webhook_hash
        {
          message: @message.webhook_hash,
          status: status,
          details: details,
          output: "", #<--- replace output with an empty string
          sent_with_ssl: sent_with_ssl,
          timestamp: @attributes["timestamp"],
          time: time
        }
      end`

this works (i don't use any webhooks)... 
but in the end i switched back to version 2.1.2

**Comment by catphish** [comment]
This should be fixed by f0a8aca6e10064fb16daefff9e22dcc20a831868
